service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -261,6 +263,24 @@ private boolean compareAndReRegisterSchema(Entry<String, String> localSchemaEntr
       String scSchemaContent = srClient.getSchema(microservice.getServiceId(), scSchema.getSchemaId());
       String localSchemaContent = localSchemaEntry.getValue();
 
+      if (!StringUtils.isEmpty(scSchemaContent) && !StringUtils.isEmpty(localSchemaContent)) {
+        Swagger scSwagger = SwaggerUtils.parseSwagger(scSchemaContent);
+        Swagger localSwagger = SwaggerUtils.parseSwagger(localSchemaContent);
+        if (scSwagger.equals(localSwagger)) {
+          return true;
+        }
+        if (ServiceRegistryConfig.INSTANCE.isIgnoreSwaggerDifference()) {
+          LOGGER.warn(
+              "service center schema and local schema both are different:\n service center schema:\n[{}]\n local schema:\n[{}]"
+                  + "\nYou need to increment microservice version before deploying. "
```

Comment:
service center schema and local schema both are different:\n service center schema:\n[{}]\n local schema:\n[{}]"
                  + "\nYou have configured to ignore difference check. It's recommented  to increment microservice version before deploying when shcema change.

Reply:
ok